### PR TITLE
Added explanation about masking outputs

### LINKED ISSUE #16274: Masked outputs are not passed and are considered as secrets
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions

### What part(s) of the article would you like to see updated?

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#masking-a-value-in-log

### Additional information

There were multiple issues in the actions/runner repository ([1061](https://github.com/actions/runner/issues/1061),[1498](https://github.com/actions/runner/issues/1498)) and the community [discussion](https://github.community/t/skip-output-aws-account-id-since-it-may-contain-secret/123791) related to this update.

### PR DESCRIPTION
**Pull request by nikola-jokic**
### Why:

Closes #16274 

### What's being changed:
Proposed an explanation that you should not use add-mask with outputs.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@nikola-jokic 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/using-workflows/workflow-commands-for-github-actions.md [hunk 1]
```diff
@@ -190,7 +190,7 @@ echo "::endgroup::"
 ::add-mask::{value}
 ```
 
-Masking a value prevents a string or variable from being printed in the log. Each masked word separated by whitespace is replaced with the `*` character. You can use an environment variable or string for the mask's `value`.
+Masking a value prevents a string or variable from being printed in the log. Each masked word separated by whitespace is replaced with the `*` character. You can use an environment variable or string for the mask's `value`. Masked outputs are considered as secrets, and will be redacted on the runner. See [Workflow commands for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions).
```
Comment (lucascosti) [inline content/actions/using-workflows/workflow-commands-for-github-actions.md:193]: Thanks for this clarification, @nikola-jokic. I think we can add a bit more explanation as to what being 'considered as secrets' means. Is this rewording accurate?

```suggestion
Masking a value prevents a string or variable from being printed in the log. Each masked word separated by whitespace is replaced with the `*` character. You can use an environment variable or string for the mask's `value`. When you mask a value, it is treated as a secret and will be redacted on the runner. For example, after you mask a value, you won't be able to set that value as an output.
```

Comment (nikola-jokic) [reply content/actions/using-workflows/workflow-commands-for-github-actions.md:312]: Hi @lucascosti, you are right. I am committing your changes. Thank you!

**Review by lucascosti — APPROVED** [review]
Awesome! Marking this as ready to merge when the current deployment freeze is finished.

**Comment by Hmnibsheer** [comment]
Hot

**Pull request merged**